### PR TITLE
Don't issue any "not found" warnings when `-AstubNoWarnIfNotFound` is passed

### LINKED ISSUE #5390: stubNoWarnIfNotFound does not hide "Type not found" warnings
Using the gradle plugin, even when `-AstubNoWarnIfNotFound=true` has been provided, checkerframework still warns about missing stub types:

```
> Task :compileJava
warning: /home/roded/dev/games/checkerframework-stubNoWarnIfNotFound/checkerframework/kr.pe.kwonnam.slf4jlambda.LambdaLogger.astub:(line 5,col 1): Type not found: kr.pe.kwonnam.slf4jlambda.LambdaLogger
1 warning
```

I'll soon attach a reproduction project.

Edit: A project reproducing the issue can be found here: 
https://github.com/roded/checker-framework-issue-5390

Clone and run `./gradlew build`:
```
[roded@ryzen checker-framework-issue-5390]$ ./gradlew build

Welcome to Gradle 7.4!

Here are the highlights of this release:
 - Aggregated test and JaCoCo reports
 - Marking additional test source directories as tests in IntelliJ
 - Support for Adoptium JDKs in Java toolchains

For more details see https://docs.gradle.org/7.4/release-notes.html

Starting a Gradle Daemon, 2 stopped Daemons could not be reused, use --status for details

> Task :compileJava
warning: /home/roded/dev/games/checker-framework-issue-5390/checkerframework/kr.pe.kwonnam.slf4jlambda.LambdaLogger.astub:(line 5,col 1): Type not found: kr.pe.kwonnam.slf4jlambda.LambdaLogger
1 warning

BUILD SUCCESSFUL in 8s
3 actionable tasks: 3 executed
```

### PR DESCRIPTION
**Pull request by smillst**
Fixes #5390 .

### CODE REVIEW

File: framework/src/main/java/org/checkerframework/framework/stub/AnnotationFileParser.java [hunk 1]
```diff
@@ -2908,7 +2908,7 @@ private void stubWarnNotFound(NodeWithRange<?> astNode, String warning) {
    * @param warnIfNotFound whether to print warnings about types/members that were not found
    */
   private void stubWarnNotFound(NodeWithRange<?> astNode, String warning, boolean warnIfNotFound) {
-    if ((fileType.isCommandLine() || warnIfNotFound) || debugAnnotationFileParser) {
```
Comment (mernst) [inline framework/src/main/java/org/checkerframework/framework/stub/AnnotationFileParser.java:2911]: Why are the parentheses needed?  I think they can be dispensed with.

**Review by mernst — APPROVED** [review]
Thanks!

File: framework/src/main/java/org/checkerframework/framework/stub/AnnotationFileParser.java (hunk 1 above)
Comment (smillst) [reply framework/src/main/java/org/checkerframework/framework/stub/AnnotationFileParser.java:2911]: What parentheses?

Comment (smillst) [reply framework/src/main/java/org/checkerframework/framework/stub/AnnotationFileParser.java:2911]: You are looking at the old line of code. The new line is (warnIfNotFound) || debugAnnotationFileParser)

**Pull request merged**